worker/src/synchronizer.rs

```diff
@@ -133,55 +136,80 @@ impl Synchronizer {
                 // Handle primary's messages.
                 Some(message) = self.rx_message.recv() => match message {
                     PrimaryWorkerMessage::Synchronize(digests, target) => {
-                        let now = SystemTime::now()
-                            .duration_since(UNIX_EPOCH)
-                            .expect("Failed to measure time")
-                            .as_millis();
+                        let mut missing = HashSet::new();
+                        let mut available = HashSet::new();
 
-                        let mut missing = Vec::new();
-                        for digest in digests {
+                        for digest in digests.iter() {
                             // Ensure we do not send twice the same sync request.
-                            if self.pending.contains_key(&digest) {
+                            if self.pending.contains_key(digest) {
                                 continue;
                             }
 
                             // Check if we received the batch in the meantime.
-                            match self.store.read(digest).await {
+                            match self.store.read(*digest).await {
                                 Ok(None) => {
-                                    missing.push(digest);
+                                    missing.insert(*digest);
                                     debug!("Requesting sync for batch {digest}");
                                 },
                                 Ok(Some(_)) => {
                                     // The batch arrived in the meantime: no need to request it.
+                                    available.insert(*digest);
+                                    debug!("Digest {digest} already in store, nothing to sync");
```

Comment:
This may end up being quite verbose, do we need this log at any level above `trace`?

Reply:
Yeah I know it will be verbose, but given the current behaviour in DevNet I err on the side of having extra logs to be able to trace any related issues - it was a bit hard to diagnose with the current level of logging. That being said though, I'll change this to `trace` and we can enable in worker the trace level - and switch back if we don't need to any more.